### PR TITLE
Clean up cc script

### PR DESCRIPTION
**Pull request by eschnett**
This is just a cleanup, no functionality should change.

### CODE REVIEW

**Comment by adamjstewart** [comment]
While you're at it, would you mind converting all single equals sign equality to double equals sign?

File: lib/spack/env/cc [hunk 1]
```diff
@@ -130,11 +115,11 @@ case "$command" in
...
-if [ -z "$mode" ]; then
+# vcheck mode, nothing is added in terms of extra search paths or
+# libraries.
+if [[ -z $mode ]]; then
     for arg in "$@"; do
-        if [ "$arg" = -v -o "$arg" = -V -o "$arg" = --version -o "$arg" = -dumpversion ]; then
+        if [[ $arg = -v || $arg = -V || $arg = --version || $arg = -dumpversion ]]; then
             mode=vcheck
             break
     fi
```
Comment (adamjstewart) [inline lib/spack/env/cc]: This line should be indented more.

Comment (eschnett) [inline lib/spack/env/cc]: That was probably a tab instead of a space. I removed all tabs.

**Comment by eschnett** [comment]
Comparisons converted.

**Comment by adamjstewart** [comment]
I think the first line should be `#!/usr/bin/env bash` just like other scripts use `#!/usr/bin/env python`. On OS X, I don't use the builtin bash because it's too old.

Although that might slow this script down because it's called so many times... On second thought, don't worry about it.

File: lib/spack/env/cc [hunk 2]
```diff
@@ -130,11 +115,11 @@ case "$command" in
 esac
 
 # If any of the arguments below is present then the mode is vcheck. In
```
Comment (adamjstewart) [inline lib/spack/env/cc]: _are_ present

**Comment by adamjstewart** [comment]
This looks a lot better now. I have some rpath fixes for NAG in store so I'll wait until this gets merged and rebase off of it.

File: lib/spack/env/cc [hunk 3]
```diff
@@ -66,11 +66,10 @@ function die {
 
 for param in $parameters; do
     if [[ -z ${!param} ]]; then
-        die "Spack compiler must be run from spack!  Input $param was missing!"
+        die "Spack compiler must be run from Apack! Input '$param' is missing."
```
Comment (tgamblin) [inline lib/spack/env/cc]: Apack?

**Comment by tgamblin** [comment]
Looks good except `Apack`.  

I don't mind using `#!/usr/bin/env bash` if it's necessary, but I have yet to see a system I care about that doesn't have `/bin/bash`.

If we *have* to port to a non-bash system one day, we might have to rewrite `cc` in POSIX-compliant `sh`.  Worth noting that Debian uses `dash` for its `/bin/sh`, and others I think use `ash`. Again, though, I've never seen a user-facing Debian install w/o bash.

**Comment by eschnett** [comment]
Withdrew secret attempt to rename project to `Apack`.

**Comment by adamjstewart** [comment]
Looks good to me now. Approved.

**Comment by gartung** [comment]
When invoked from a makefile with /bin/sh bash is put in posix mode. 

Use of [[ ]] breaks posix compliance. See

http://wiki.bash-hackers.org/scripting/nonportable

This is the likely cause of 
https://github.com/LLNL/spack/issues/809

This might impact other systems with posix compliant bash.

**Pull request merged**